gap/PolygonalComplexes/properties.gi

```diff
@@ -218,6 +218,159 @@ InstallMethod( FaceCounter, "for a twisted polygonal complex",
         return Collected( Compacted( faceDegrees ) );
     end
 );
+
+InstallMethod( ButterflyCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local e,temp2,counter,g,VerticesOfOrthogonalEdge,voe1,voe2,degOfVert1,degOfVert2;
+
+        VerticesOfOrthogonalEdge:=function(S,e)
+            local temp;
+            temp:=FacesOfEdge(S,e);
+            temp:=Union(VerticesOfFace(S,temp[1]),VerticesOfFace(S,temp[2]));
+            return Difference(temp, VerticesOfEdge(S,e));
+	end;
+	counter:=[];
+	for e in InnerEdges(surface) do 
+		voe1:=VerticesOfEdge(surface,e);
+		degOfVert1:=SortedList(List(voe1,g->FaceDegreeOfVertex(surface,g)));
+		voe2:=VerticesOfOrthogonalEdge(surface,e);
+		degOfVert2:=SortedList(List(voe2,g->FaceDegreeOfVertex(surface,g)));
+		Add(counter,[degOfVert1,degOfVert2]);
+	od;
+	return Set(Collected(counter));
+end
+);
+
+
+InstallMethod( UmbrellaCounter, "for a closed simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local n,counter,v,temp,orb,tup,G,perm,OrbitOnList,tempcounter,
+	verticesOfUmb,edgesOfUmb,facdegOfVert,temp1,EquivalentLists,
+	ComputeSmallestRepresentative,bub,equi,help_SortCounter;
+
+	if not IsClosedSurface(surface) then 
+		return fail;
+	fi;
+        #facdegList is a list of positive integers describing the face degrees
+        #of the boundary vertices of a closed umbrella, whereby the degrees
+        #of two neighbouring vertices of the umbrella stand next to each
+	#other in the given list. Since there are different lists with the
+	#same property describing the same closed umbrella, this function
+	#returns all those lists. 
+        EquivalentLists:=function(G,facdegList)
+                local g,temp,i,EquivFacDegLists;
+                EquivFacDegLists:=[];
+                for g in G do
+                        temp:=[];
+                        for i in [1..Length(facdegList)] do
+                                temp[i]:=facdegList[i^g];
+                        od;
+                        Add(EquivFacDegLists,temp);
+                od;
+                return EquivFacDegLists;
+        end;
+
+
+
+	#As mentioned before, there are different lists of integers describing
+	#the same closed umbrella. This functions returns the lexicographically
+	#smallest list of all the possibilities.
+	ComputeSmallestRepresentative:=function(L)
+		local min,tempL,i;
+		tempL:=L;		
+		i:=2;
+		min:=Minimum(tempL[1]);
+		tempL:=Filtered(tempL,g->g[1]=min);
+		while tempL <>[] and i<=Length(tempL[1]) do 
+			min:=Minimum(List(tempL,g->g[i]));
+			tempL:=Filtered(tempL,g->g[i]=min);
+			i:=i+1;
+			if Length(tempL)=1 then
+				return tempL[1];
+			fi; 
+		od;
+		return tempL[1];
+	end;
+
+
+	#The list tcounter consists of lists [list1,n] where list1 is a list of 
```

Comment:
In my opinion the description is wrong. I think tcounter is just list 1 and not [list1, n]. Because when you call this method in line 334, you only give the method the first entry of the elements in counter.